### PR TITLE
CI: fix pre-release job that is failing on Cython 3.0b1

### PR DESCRIPTION
**Pull request by rgommers**
by pinning to `<3.0b1`. xref gh-17234 for the remaining issues

[skip actions] [skip cirrus] [skip circle]

### CODE REVIEW

File: ci/azure-travis-template.yaml [hunk 1]
```diff
@@ -70,7 +70,7 @@ steps:
     pip install --upgrade ${{parameters.numpy_spec}} &&
     pip install --upgrade pip setuptools==59.6.0 wheel build meson meson-python &&
     pip install ${{parameters.other_spec}}
-    cython
+    cython<3.0b1  # unpin when gh-17234 is addressed
```
Comment (tylerjereddy) [inline ci/azure-travis-template.yaml:73]: checking the error locally, may just need quotes `"cython<3.0b1"`

Comment (tylerjereddy) [reply ci/azure-travis-template.yaml:73]: I'll push it in

**Comment by tylerjereddy** [comment]
Hmm, `ModuleNotFoundError: No module named 'mpmath'`

This may be because the comment seems to get squished into the pip command:

> "cython<3.0b1"  # unpin when gh-17234 is addressed gmpy2 threadpoolctl

because of the way the yml gets condensed with `>-`

let me try removing the comment..

**Comment by rgommers** [comment]
That worked, the job is green now and it uses `cython-3.0.0a11`. So I'll get this in.

**Comment by rgommers** [comment]
Thanks for the help @tylerjereddy

**Pull request merged**